src/model/MultiValueColumn.ts

```diff
@@ -86,151 +89,130 @@ export default class MultiValueColumn extends ValueColumn<number[] > {
-        .range([this.colorrange[0], 'white', this.colorrange[1]]);
+        .range([this.colorRange[0], 'white', this.colorRange[1]]);
 
     } else {
       this.colorScale
         .domain([this.min, this.max])
-        .range(['white', this.colorrange[1]]);
+        .range(['white', this.colorRange[1]]);
     }
   }
 
 
-  private boxPlotWidth() {
-    this.boxPlotScale
-      .domain([this.min, this.max])
-      .range([0, this.getWidth()])
-
-  }
-
-
   compare(a: any, b: any, aIndex: number, bIndex: number) {
-    this.sortBy = this.getUserSortBy();
-    const a_val = this.getValue(a, aIndex);
-    const b_val = this.getValue(b, bIndex);
+
+    const a_val = (this.getValue(a, aIndex)).sort(numSort);
+    const b_val = (this.getValue(b, bIndex)).sort(numSort);
     const sort: any = new CustomSortCalculation(a_val, b_val);
     const f = sort[this.sortBy].bind(sort);
 
     return f();
   }
 
+  getColor() {
 
-  getLabel(row: any, index: number) {
-    return '' + this.getValue(row, index);
-  }
-
-  getRaw(row: any, index: number) {
-    return this.accessor(row, index, this.id, this.desc, this.findMyRanker());
-  }
-
-  getValue(row: any, index: number) {
-    var v = this.getRaw(row, index);
-    return (v);
+    return this.colorScale;
   }
 
-  getColor(data: any) {
+  calculateCellDimension(width) {
 
-    return this.colorScale(data);
-  }
-
-  calculateCellDimension() {
 
-    return (this.getWidth() / this.bins);
+    return (width * 1 / this.dataLength);
   }
 
-  getxScale(data) {
+  getSparkLineXScale() {
     this.xposScale
-      .domain([0, this.bins - 1])
-      .range([0, this.getWidth()]);
-    return this.xposScale(data);
+      .domain([0, this.dataLength - 1]);
+    return this.xposScale;
   }
 
-  getyScale(data, rowheight) {
+  getSparkLineYScale() {
     this.yposScale
-      .domain([this.min, this.max])
-      .range([rowheight, 0]);
-    return this.yposScale(data);
+      .domain([this.min, this.max]);
+    return this.yposScale;
   }
 
+  getDataLength() {
 
-  getbinaryColor() {
-
-    return this.colorrange;
+    return this.dataLength;
   }
 
-  getthresholdValue() {
-
-    return this.threshold;
-  }
+  getbinaryColor() {
 
-  getVerticalBarHeight(data, rowheight) {
 
-    if (this.min < this.threshold) {
-      this.verticalBarScale
-        .domain([this.min, this.max])
-        .range([0, rowheight / 2]);
+    return this.colorRange;
+  }
 
-      this.verticalBarHeight = (rowheight / 2 - this.verticalBarScale(data));
+  getDataInfo() {
 
-    } else {
-      this.verticalBarScale
-        .domain([this.min, this.max])
-        .range([0, rowheight]);
+    return this.dataInfo;
+  }
 
-      this.verticalBarHeight = this.verticalBarScale(data);
-    }
+  getVerticalBarScale() {
+    this.verticalBarScale
+      .domain([this.min, this.max]);
+    return this.verticalBarScale;
+  }
 
-    return this.verticalBarHeight;
+  getboxPlotScale(width) {
+    this.boxPlotScale
```

Comment:
this getter manipulates the own `boxPlotScale` field on each call based on the given argument that is just valid for one render call. I would avoid that, since you never really know what the current state of the scale in the model is